### ISSUE TITLE
Integrated Gradients - Higher Convergence Delta with more Steps? 

### ISSUE THREAD

**Issue by Steven-Palayew** [pytorch/captum#1334]
Working with the example from this tutorial here: https://github.com/munnm/XAI-for-practitioners/blob/462b1fc79d9cf7998992e1878c60d9d4c6282982/05-text/layer_integrated_gradients.ipynb, it seems as though using less steps actually _decreases_ the size of the convergence delta? Ie see screenshots below. 

<img width="468" alt="image" src="https://github.com/user-attachments/assets/78336e7d-324c-4c48-88d2-366359755d48">

<img width="468" alt="image" src="https://github.com/user-attachments/assets/5669acf4-3be3-4529-a4e0-4856832b1c49">

<img width="468" alt="image" src="https://github.com/user-attachments/assets/863dcad4-db23-4c3c-9d34-f25b35bef97b">

I'm just wondering if this is normal behaviour? From #311 it seems like we would expect the opposite to occur. 

TIA!

**Comment by craymichael** [comment]
Apologies for the delay.

It can happen. The convergence delta is a numerical estimate of how well the attribution sum satisfies completeness:

```text
sum(attributions) ~= F(input) - F(baseline)
```

Increasing `n_steps` usually helps, but the delta is not guaranteed to decrease monotonically for every model and every integration method. Non-monotonic behavior is especially possible for NLP models because the straight-line path between baseline embeddings and input embeddings can pass through regions the model was never trained on.

Things worth checking:

- call `model.eval()` before attribution,
- attribute logits rather than softmax probabilities when possible,
- use a meaningful baseline token such as PAD / mask rather than all zeros if the model expects token embeddings,
- compare `sum(attributions)` directly to `F(input) - F(baseline)`,
- try both `method="gausslegendre"` and a Riemann method,
- make sure `internal_batch_size` is not changing model behavior through batch-dependent layers.

A lower delta at fewer steps can be a numerical coincidence from cancellation, so I would not treat one lower-delta run as proof that fewer steps is generally better.

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
